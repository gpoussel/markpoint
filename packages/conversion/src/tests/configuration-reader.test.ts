import { readdir } from 'node:fs/promises'
import { join } from 'node:path'

import { describe, it } from 'vitest'

import { TemplateConfigurationReader } from '..'

describe('TemplateConfigurationReader', () => {
  const templateConfiguration = new TemplateConfigurationReader()
  describe('readConfiguration', async () => {
    const validFixtures = join(__dirname, '__fixtures__', 'valid')
    const validFiles = await readdir(validFixtures)
    for (const file of validFiles) {
      it(`should read the valid/${file} configuration file and return the parsed content`, async ({ expect }) => {
        const result = await templateConfiguration.readConfiguration(join(__dirname, '__fixtures__', 'valid', file))

        // Update result to avoid snapshot mismatches due to different file paths
        expect(result.baseFile).toMatch(/example\.pptx$/)
        result.baseFile = '../example.pptx'

        expect(result).toMatchSnapshot()
      })
    }

    const invalidFixtures = join(__dirname, '__fixtures__', 'invalid')
    const invalidFiles = await readdir(invalidFixtures)
    for (const file of invalidFiles) {
      it(`should read the invalid/${file} configuration file and throw an error`, async ({ expect }) => {
        await expect(() =>
          templateConfiguration.readConfiguration(join(__dirname, '__fixtures__', 'invalid', file)),
        ).rejects.toThrow()
      })
    }
  })
})
