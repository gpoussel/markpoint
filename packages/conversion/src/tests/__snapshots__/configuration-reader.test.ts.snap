// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TemplateConfigurationReader > readConfiguration > should read the valid/simple.yml configuration file and return the parsed content 1`] = `
{
  "baseFile": "../example.pptx",
  "layouts": [
    {
      "baseSlideNumber": 4,
      "elements": [
        {
          "creationId": "{6D871570-FBB8-F282-1B97-65C16AE6C6F4}",
          "name": "title",
          "type": "line",
        },
        {
          "creationId": "{4F18DD32-924E-CE9C-91B0-4F30CCE1B6F9}",
          "name": "subtitle",
          "type": "line",
        },
      ],
      "name": "title",
    },
    {
      "baseSlideNumber": 24,
      "elements": [
        {
          "creationId": "{05383CED-A871-D722-5C38-FDB6D14AA784}",
          "name": "title",
          "type": "line",
        },
        {
          "creationId": "{4F2608E1-5E03-2C7E-434E-5BF07E81F24A}",
          "name": "subtitle",
          "type": "line",
        },
        {
          "creationId": "{7B4224E4-95EA-4DE2-B734-333DC4BF7A4F}",
          "name": "background",
          "type": "picture",
        },
      ],
      "name": "sectionTitle",
    },
    {
      "baseSlideNumber": 21,
      "elements": [
        {
          "creationId": "{4582957B-1559-50B8-87D1-3F29F9DD12B2}",
          "name": "title",
          "type": "line",
        },
        {
          "creationId": "{28862325-59A6-26E6-FB10-93F6B61575F0}",
          "name": "subtitle",
          "type": "line",
        },
        {
          "creationId": "{0984FDF1-E23B-9E8F-73B5-6BF495603FA0}",
          "name": "content",
          "type": "text",
        },
      ],
      "name": "contentSlide",
    },
    {
      "baseSlideNumber": 63,
      "elements": [],
      "name": "end",
    },
  ],
  "master": {
    "elements": [
      {
        "creationId": "{62086BA9-1591-C6CD-E573-A5F8961683B1}",
        "name": "footer",
        "type": "line",
      },
    ],
  },
}
`;
